numba/tests/test_fancy_indexing.py

```diff
@@ -3,206 +3,163 @@
+
+        # Boolean Arrays
+        ((5, 6, 7, 8, 9, 10), (slice(4, 5), 3,
+                               np.array([True, False, True, False, True, False, False]),
+                               1)),
+        ((5, 6, 7, 8, 9, 10), (3, np.array([True, False, True, False, True, False]),
+                               slice(None), slice(4))),
+
+        # Pure Arrays
+        ((5, 6, 7, 8, 9, 10), np.array([0,3,-2], dtype=np.int16)),
+        ((5, 6, 7, 8, 9, 10), np.array([0,3,1], dtype=np.uint16)),
+        ((5, 6, 7, 8, 9, 10), np.array([False,True,True,False,False])),
+    ]
+
+    def check_getitem_indices(self, arr_shape, index):
+        def get_item(array, idx):
+            return array[index]
+
+        arr = np.random.random_integers(0, 10, size=arr_shape)
+        get_item_numba = njit(get_item)
         orig = arr.copy()
         orig_base = arr.base or arr
 
-        for index in indices:
-            expected = pyfunc(arr, index)
-            # Sanity check: if a copy wasn't made, this wasn't advanced
-            # but basic indexing, and shouldn't be tested here.
-            assert expected.base is not orig_base
-            got = cfunc(arr, index)
-            # Note Numba may not return the same array strides and
-            # contiguity as Numpy
-            self.assertEqual(got.shape, expected.shape)
-            self.assertEqual(got.dtype, expected.dtype)
-            np.testing.assert_equal(got, expected)
-            # Check a copy was *really* returned by Numba
-            if got.size:
-                got.fill(42)
-                np.testing.assert_equal(arr, orig)
-
-    def test_getitem_tuple(self):
-        # Test many variations of advanced indexing with a tuple index
-        N = 4
-        ndim = 3
-        arr = np.arange(N ** ndim).reshape((N,) * ndim).astype(np.int32)
-        indices = self.generate_advanced_index_tuples(N, ndim)
-
-        self.check_getitem_indices(arr, indices)
-
-    def test_getitem_tuple_and_ellipsis(self):
-        # Same, but also insert an ellipsis at a random point
-        N = 4
-        ndim = 3
-        arr = np.arange(N ** ndim).reshape((N,) * ndim).astype(np.int32)
-        indices = self.generate_advanced_index_tuples_with_ellipsis(N, ndim,
-                                                                    many=False)
-
-        self.check_getitem_indices(arr, indices)
+        expected = get_item(arr, index)
+        got = get_item_numba(arr, index)
+        # Sanity check: In advanced indexing, the result is always a copy.
+        assert expected.base is not orig_base
+
+        # Note: Numba may not return the same array strides and
+        # contiguity as Numpy
+        self.assertEqual(got.shape, expected.shape)
+        self.assertEqual(got.dtype, expected.dtype)
+        np.testing.assert_equal(got, expected)
+
+        # Check a copy was *really* returned by Numba
+        if got.size:
+            got.fill(42)
+            np.testing.assert_equal(arr, orig)
```

Comment:
I think you can remove the if condition here.

Reply:
Removed it.